frontend/packages/core/src/NPS/feedback.tsx

```diff
@@ -0,0 +1,281 @@
+    },
+  };
+
+  const textFieldProps = {
+    fullWidth: true,
+    InputProps: {
+      rows: 3,
+      rowsMax: 3,
+    },
+    style: {
+      marginTop: "15px",
+    },
+  };
+
+  // Will fetch the survey results for the given origin on load
+  React.useEffect(() => {
+    let data: IClutch.feedback.v1.ISurvey = defaults;
+
+    client
+      .post("/v1/feedback/getSurveys", {
+        origins: [opts.origin],
+      })
+      .then(response => {
+        const surveyData: IClutch.feedback.v1.IGetSurveysResponse = response?.data?.originSurvey;
+
+        data = surveyData[opts.origin] ?? defaults;
+      })
+      .catch((err: ClutchError) => {
+        // eslint-disable-next-line no-console
+        console.error(err);
+      })
+      .finally(() => {
+        setSurvey(data);
+      });
+  }, []);
+
+  // Will debounce feedback requests to the server in case of multiple quick changes to selected
+  const sendFeedback = React.useCallback(
+    debounce((formData: IClutch.feedback.v1.ISubmitFeedbackRequest) => {
+      client
+        .post("/v1/feedback/submitFeedback", { userId: userId(), ...formData })
+        .catch((err: ClutchError) => {
+          // eslint-disable-next-line no-console
+          console.error(err);
+        });
+    }, debounceTimer),
+    []
+  );
+
+  // On a change to submit or selected will attempt to send a feedback request
+  React.useEffect(() => {
+    if (selected) {
+      sendFeedback({
+        id: requestId,
+        feedback: {
+          ratingLabel: selected.label,
+          ratingScale: {
+            emoji: IClutch.feedback.v1.EmojiRating[selected.emoji],
+          },
+          urlPath: window.location.pathname,
+          freeformResponse: trimmed,
+        },
+        metadata: {
+          origin: IClutch.feedback.v1.Origin[opts.origin],
+          userSubmitted: hasSubmit,
+          survey,
+        },
+      });
+    }
+  }, [selected, hasSubmit]);
+
+  // Form onSubmit handler
+  const submitFeedback = e => {
+    if (e) {
+      e.preventDefault();
+    }
+    setSubmit(true);
+    if (opts.onSubmit) {
+      opts.onSubmit(true);
+    }
+  };
+
+  return (
+    <>
+      {hasSubmit ? (
+        <StyledAlert severity="info" {...AlertProps}>
+          <Text>Thank you for your feedback!</Text>
+        </StyledAlert>
+      ) : (
+        <form onSubmit={submitFeedback}>
+          <StyledGrid container direction="row" alignItems="center" justify="center">
+            <MuiGrid item xs={6}>
+              <Text>{survey.prompt}</Text>
+            </MuiGrid>
+            <MuiGrid item xs={6} style={{ display: "flex", justifyContent: "space-around" }}>
+              <EmojiRatings ratings={survey.ratingLabels} setRating={setSelected} />
+            </MuiGrid>
+            {selected !== null && (
+              <>
+                <MuiGrid item xs={12}>
+                  <TextField
+                    multiline
+                    placeholder={survey.freeformPrompt}
```

Comment:
can we override the TextField styling in this file to make the fontSize 14px? jst b/c it looks too big in proportion to everything else

Reply:
Yup, overrode to "14px"